SafeguardSessions/modules/version/Version.pqm

```diff
@@ -25,8 +26,14 @@ let
             version,
     Version.CheckVersion = (version as text, supportedVersions as list) as logical =>
         let
+            // TODO: Text.Split by ".", List.Zip, Compare elements
             IsVersionAccepted = (version, supportedVersion) =>
-                Text.StartsWith(version, supportedVersion, Comparer.OrdinalIgnoreCase),
+                let
+                    versionParts = Text.Split(version, Version.Separator),
+                    supportedVersionParts = Text.Split(supportedVersion, Version.Separator),
+                    commonLength = List.Min({List.Count(versionParts), List.Count(supportedVersionParts)})

```

Comment:
I'd reject a version having less parts then the supported version currently examined.

Reply:
SPS version is a full version: 7.0.0, 7.3.1, etc.
Custom versions are mainly shorter: master-HEAD+20230615T101515Z

Supported version at this moment contains 7.3.

We can explicitly check for the length, but it I think it is unnecessary, as master-HEAD+20230615T101515Z will never be equal to 7.

It is only a problem if for some reason an SPS has a version of simply 7 (extra custom firmware or something like that).

But basically, instead of commonLength and List.FirstN we would have an if-then-else.

if List.Count(versionParts) < List.Count(versionParts) then false else ...

I'll think about it.

Reply:
List equality needs equal lenght lists to work properly.

So a simple if is not enough, as {7,3,0} would never be equal to {7,3}.

Common length is mainly because of the shorter form of the supported version.

The previous scenario, where the fetched version is shorter is handled, is just an aftereffect.

Reply:
My concern is with version number like: {7} getting past, since it's shorter then {7,3} supported version.
Since it's rather theoretical, I think it's ok the shorter of the two to determine the common length.